### ISSUE TITLE
Signing in with blank password fails with not_found_in_database instead of invalid

### ISSUE THREAD

**Issue by Drenmi** [plataformatec/devise#4953]
## Context

I'm trying to provide more helpful alerts to end users, which would be one of the following:

1. Incorrect password: show a message indicating the password is wrong. (`invalid`)
2. User doesn't exist: show a message indicating the email is wrong. `(not_found_in_database`)

This mostly works, except for the case where the password field is left blank, which results in the second message being displayed. This is not correct behaviour, since the user does indeed exist in the database.

The issue is possibly on [this line](https://github.com/plataformatec/devise/blob/715192a7709a4c02127afb067e66230061b82cf2/lib/devise/strategies/database_authenticatable.rb#L10) where the presence of password is checked before setting `resource`.

## Environment

- Ruby: 2.5.1
- Rails: 5.2.1
- Devise: 4.5.0

## Current behavior

1. Set up an account (database authenticatable.)
2. Sign in with the correct authentication key, but leave password blank.
3. It fails with `not_found_in_database`.

## Expected behavior

It fails with `invalid`, since the account exists in database, but the password was incorrect.

**Comment by w-sanches** [comment]
Although it is useful to tell the user that his email is not in the database (many sites including Google, Microsoft, Slack and others do it), it is actually considered a [possible security issue](https://www.owasp.org/index.php/Authentication_Cheat_Sheet#Authentication_and_Error_Messages).
Knowing the email is valid, but the password is not, would allow an attacker to brute force the password.

**Comment by Drenmi** [comment]
Yes. I am aware of this argument. It is, however, undoubtedly a case of "security through obscurity" (in this case at the expense of the legitimate user), and I think the recommendation should be to guard against brute force attacks in general, instead of hoping that the attacker doesn't know which account they are attacking. 😅
